### ISSUE TITLE
On download page, also offer 0install compiled to a .jar using  "Ocamljava"

### ISSUE THREAD

**Issue by zaxebo1** [0install/0install#40]
**OcamlJava ( http://www.ocamljava.org/ ) implements 100% of ocaml 4.0 in pure java  ,  so a application written in ocaml can be compiled to java .jar using ocamljava**
the homepage of ocamljava says " 100% OCaml, 100% Java. The compiler supports the whole OCaml language (version 4.01.0), and generates plain Java bytecode. Moreover, the runtime support library does not use native code. Produced jar files can thus be run on any 1.7+ JVM."

****

http://0install.net/injector.html and http://0install.net/install-linux.html gives separate "binary" compile  downloads of 0install injector.

I propose that as 0install 2.0 has been written in ocaml , hence it may be compiled with ocamljava and turned into a .jar which can be run on any jvm. And now this 0install-injector.jar can be also given as "universal binary" which can run on any platform with JRE . 

I am not asking that other downloads be removed for platform specific binary of 0install, but I am just saying that you can provide a universal .jar for those who are not speed-conscious ,butwant to deploy a same binary , and also if their platform bundles old version of 0install due to inactive package maintenance, then they will have atleast a fallback to try latest version of 0install-injector.jar, instead of old version of the packaged 0install-injector (native)

**Comment by zaxebo1** [comment]
Due to a hude dependency list (shown in "ldd 0install" ) it is difficult to take this 0install binary to various systems(with various linux distributions having various old versions) in safe way without fear of crash.
So atleast .jar of 0install (via ocamljava) can serve as a **fallback**, even if speed may be slow compared to natively compiled version

**Comment by afb** [comment]
What Linux distribution (and release) are you targeting ? Does it have Java 7 available ?

**Comment by zaxebo1** [comment]
multiple production servers to manage across **multiple geographical locations** having various linux distributions -slackware 13., ubuntu 12.04 server  , Rocks with CentOS 5  ,  Arch etc.

Rocks cluster with CentOS  was installed in 2007 , is supported till 2017.
Slackware 13.1 was installed in 2010.
Ubuntu 12.04 server installed in 2012
All have openJDK 1.7 (some compiled from source; some from binary ) installed due to some java applications running on them. 

All have different versions of .so and native libs versions to depend upon, but JDK 1.7 is something which we can depend upon, as it is guaranteed to be there on each server as per policy. And this scene is there in quiet number of organizations.
The testing servers replicate the above mentioned production servers in exactly same OS versions. 


I would prefer to have common layer of distribution channel using zeroinstall for our testing servers, it is good to have zeroinstall running on JVM(via ocamljava.org) . So that I do not have to depend upon various Gtk and other .so versions. The same ocaml source code of zeroinstall , compiled to jvm ( and may be second GUI in javafx/swing) (Apart from Gtk based GUI) - this will be most ideal.

**Comment by afb** [comment]
Sure, it sounds like an interesting option/alternative... Let us know how it goes, when you have installed ocamljava and ocj-ocamlfind and managed to compile/convert the regular 0install and the opam deps ?

**Comment by talex5** [comment]
You'll also need to replace libcurl (C bindings) with cohttp (pure OCaml). That would be nice as an option anyway (for the generic binaries), as libcurl isn't always present. Should be easy enough, as it's only used in one place I think.

**Comment by afb** [comment]
@talex5: yeah, that sounds like a nice feature. couldn't get the rest to build either, though...
Seems like the ocamljava project is still in "alpha", and the OPAM integration is a bit tricky
